pipeline_dp/pipeline_operations.py

```diff
@@ -322,4 +323,14 @@ def count_per_element(self, col, stage_name: typing.Optional[str] = None):
         yield from collections.Counter(col).items()
 
     def reduce_accumulators_per_key(self, col, stage_name: str = None):
-        raise NotImplementedError()
+        def merge_accumulators_generator():
+            result = {}
+            for key, acc in col:
+                if key in result:
+                    result[key].add_accumulator(acc)
+                else:
+                    result[key] = acc
+            for item in result.items():
+                yield item
+
+        return merge_accumulators_generator()
```

Comment:
Nit: we can reuse existing API: 
self.map(self.group_by_key(), accumulator.merge)

(it's needed to import accumulator)

Reply:
Did a slightly modified version of what you suggested to get it working. Let me know what you think.

Reply:
ah, true, `map`, wouldn't work. Would
map_**values**(self.group_by_key(), accumulator.merge) work?

Reply:
Yup that simplified it a lot. Take a look now.

Reply:
It looks like we can simplify even more, namely

`self.map_values(self.group_by_key(col), accumulator.merge) `
is already implementation of reduce_per_key, i.e. we don't need `def merge_accumulators_generator():`

or I miss something?

Reply:
Oh no you're right I was missing that. Updated now.